src/repo/version.js

```diff
@@ -0,0 +1,48 @@
+'use strict'
+
+const errors = require('../errors')
+const Datastore = require('datastore-fs')
+
+const Key = require('interface-datastore').Key
+
+const versionKey = new Key('version')
+
+exports.getVersion = getVersion
+
+/**
+ * Function that has responsibility to retrieve version of repo from its root datastore's instance.
+ * This function needs to be cross-repo-version functional to be able to fetch any version number,
+ * even in case of change of repo's versioning.
+ *
+ * @param {string} path
+ * @returns {Promise<int>}
+ */
+async function getVersion (path) {
+  const store = new Datastore(path, { extension: '', createIfMissing: false })
+  await store.open()
+
+  if (!await store.has(versionKey)) {
+    throw new errors.UnknownRepoStructureError('Repo does not have version file! Is the repo initialized?')
+  }
+
+  const version = parseInt(await store.get(versionKey))
```

Comment:
It seems like [fs-datastore returns a Buffer](https://github.com/ipfs/js-datastore-fs/blob/master/src/index.js#L188) from `store.get()`

Should the result first be converted to a string before calling `parseInt()`?

Reply:
Well, I believe that `parseInt()` does this implicitly, so I would say no need for it. If for code clarity you think it should be explicit, the let me know.

Reply:
Interesting, ok I didn't think that through :)

Reply:
Tbh, I did not think about it before you mentioned, but [documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#Parameters) confirmed it ;-)